pkg/cmd/flags/logger.go

```diff
@@ -8,37 +8,59 @@ import (
 )
 
 func logHelp() string {
-	return `Control logger level priority.
+	return `Control logger options - aggregation and level priority.
 
 Possible options:
-  --log debug               | debug log level
-  --log info                | information log level (default) 
-  --log warn                | warning log level
-  --log error               | error log level
-  --log panic               | panic log level
+  --log aggregate              | turns log aggregation on, delaying output (default is off)
+  --log debug                  | debug log level
+  --log info                   | information log level
+  --log warn                   | warning log level
+  --log error                  | error log level (default)
+  --log panic                  | panic log level
+
+Examples:
+  --log debug                  | outputs debug level logs
+  --log debug --log aggregate  | outputs aggregated debug level logs every 3 seconds
 `
 }
 
-func PrepareLogger(logLevel string, w io.Writer) error {
-	var lvl logger.Level
-
-	switch logLevel {
-	case "debug":
-		lvl = logger.DebugLevel
-	case "info":
-		lvl = logger.InfoLevel
-	case "warn":
-		lvl = logger.WarnLevel
-	case "error":
-		lvl = logger.ErrorLevel
-	case "fatal":
-		lvl = logger.FatalLevel
-	default:
-		return fmt.Errorf("invalid log level: %s, use '--log help' for more info", logLevel)
+func PrepareLogger(logOptions []string, w io.Writer) error {
+	var (
+		agg bool
+		lvl logger.Level = logger.ErrorLevel
+	)
+
+	for _, opt := range logOptions {
+		switch opt {
+		case "aggregate":
```

Comment:
--log aggregate or --log aggregate:10 (For 10 seconds ?), etc ? I have another comment asking about the hardcoded 3 seconds.

Reply:
A similar option was implemented in the [bpf logger](https://github.com/aquasecurity/tracee/compare/8d83d67f5892dcc010fedeece00697a80c598ae1..819a130409a9f39c1812331281993110342d93c4) effort, but didn't make it into main.

Discussion: https://github.com/aquasecurity/tracee/pull/2352#discussion_r1035242599

Reply:
IMO I believe we should do it in this PR context. Not having a configurable time for the ticker seems odd to me (3 might be good for you and not for me, or for someone else).

Reply:
@rafaeldtinoco, implemented. Please, take a look.